src/normalinversechisq.jl

```diff
@@ -0,0 +1,62 @@
+# Based on Murphy (2007) Conjugate Bayesian Analysis of the Gaussian
+# distribution.  Equivalent to the Normal-Inverse Gamma under the follow
+# re-parametrization:
+#
+# m0 = μ
+# v0 = 1/κ
+# shape = ν/2
+# scale = νσ2/2
+#
+# The parameters have a natural interpretation when used as a prior for a Normal
+# distribution with unknown mean and variance: μ and σ2 are the expected mean
+# and variance, while κ and ν are the respective degrees of confidence
+# (expressed in "pseudocounts").
+
+struct NormalInverseChisq{T<:Real} <: ContinuousUnivariateDistribution
+    μ::T
+    σ2::T
+    κ::T
+    ν::T
+
+    function NormalInverseChisq{T}(μ::T, σ2::T, κ::T, ν::T) where T<:Real
+        ν >= zero(ν) &&
+            κ >= zero(κ) &&
+            σ2 > zero(σ2) ||
+            error("Variance and confidence (κ and ν) must all be positive")
+        new{T}(μ, σ2, κ, ν)
+    end
+end
+
+NormalInverseChisq() = NormalInverseChisq{Float64}(0.0, 1.0, 0.0, 0.0)
+
+function NormalInverseChisq(μ::Real, σ2::Real, κ::Real, ν::Real)
+    T = promote_type(typeof(μ), typeof(σ2), typeof(κ), typeof(ν))
+    NormalInverseChisq{T}(T(μ), T(σ2), T(κ), T(ν))
+end
+
+Base.convert(::Type{NormalInverseGamma}, d::NormalInverseChisq) =
+    NormalInverseGamma(d.μ, 1/d.κ, d.ν/2, d.ν*d.σ2/2)
```

Comment:
In the construction of the `NormalInverseChisq` type you check that the parameters are nonnegative. Perhaps you should be enforcing positivity instead since some of these guys end up as denominators.

Reply:
Being able to use 0 for the confidence parameters is important because it's an uninformative prior.  It's only really suitable for _updating_ but still important to be able to express.